dataclients/kubernetes/clusterclient.go

```diff
@@ -449,6 +459,88 @@ func (c *clusterClient) loadEndpoints() (map[definitions.ResourceID]*endpoint, e
 	return result, nil
 }
 
+// loadEndpointSlices is different from the other load$Kind()
+// functions because there are 1..N endpointslices created for a given
+// service. endpointslices need to be deduplicated and state needs to
+// be checked. We read all endpointslices and create de-duplicated
+// business objects skipperEndpointSlice instead of raw Kubernetes
+// objects, because we need just a clean list of load balancer
+// members. The returned map will return the full list of ready
+// non-terminating endpoints that should be in the load balancer of a
+// given service, check endpointSlice.ToResourceID().
+func (c *clusterClient) loadEndpointSlices() (map[definitions.ResourceID]*skipperEndpointSlice, error) {
+	var endpointSlices endpointSliceList
+
+	if err := c.getJSON(c.endpointSlicesURI+c.endpointSlicesLabelSelectors, &endpointSlices); err != nil {
+		log.Debugf("requesting all endpointslices failed: %v", err)
+		return nil, err
+	}
+
+	log.Debugf("all endpointslices received: %d", len(endpointSlices.Items))
+	mapSlices := make(map[definitions.ResourceID][]*endpointSlice)
+	for _, endpointSlice := range endpointSlices.Items {
+		resID := endpointSlice.ToResourceID() // service resource ID
+		mapSlices[resID] = append(mapSlices[resID], endpointSlice)
+	}
+
+	result := make(map[definitions.ResourceID]*skipperEndpointSlice)
+	for resID, epSlices := range mapSlices {
+		if len(epSlices) == 0 {
+			continue
+		}
+
+		result[resID] = &skipperEndpointSlice{
+			Meta: epSlices[0].Meta,
+		}
+
+		for i := range epSlices {
+			resEps := make(map[string]*skipperEndpoint)
+			terminatingEps := make(map[string]struct{})
+
+			for _, ep := range epSlices[i].Endpoints {
+				// if conditions are nil then we need to treat is as ready
+				if ep.Conditions == nil {
+					resEps[ep.Addresses[0]] = &skipperEndpoint{
+						// Addresses [1..100] of the same AddressType, as kube-proxy we use the first
+						// see also https://github.com/kubernetes/kubernetes/issues/106267
+						Address: ep.Addresses[0],
```

Comment:
Lets intriduce local variable `address := ep.Addresses[0]` with the comment and use it everywhere

Reply:
done